File: #83. Remove Duplicates from Sorted List/Soultion.py
'''
@Author: Goog Tech
@Date: 2020-07-18 21:48:22
@Description: https://leetcode-cn.com/problems/remove-duplicates-from-sorted-list/
@FilePath: \leetcode-googtech\#83. Remove Duplicates from Sorted List\Soultion.py
'''

# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution(object):
    def deleteDuplicates(self, head):
        """
        :type head: ListNode
        :rtype: ListNode
        """
        currentNode = head
        # 判断头节点是否为空
        if not head:
            return head
        # 遍历寻找及删除重复节点
        while currentNode.next:
            if currentNode.val == currentNode.next.val:
                currentNode.next = currentNode.next.next
            else:
                currentNode = currentNode.next
        # 返回头节点
        return head

File: #283. Move Zeroes/Soultion.java
/*
 * @Author: Goog Tech
 * @Date: 2020-08-14 17:42:47
 * @LastEditTime: 2020-08-14 17:59:37
 * @Description: https://leetcode-cn.com/problems/move-zeroes/
 * @FilePath: \leetcode-googtech\#283. Move Zeroes\Soultion.java
 * @WebSite: https://algorithm.show/
 */

/*
 * @lc app=leetcode.cn id=283 lang=java
 *
 * [283] 移动零
 */

// @lc code=start
class Solution {
    public void moveZeroes(int[] nums) {
        if(nums == null) return;
        //第一次遍历: 将数组中的非零元素赋值给nums[j]
        int j = 0;
        for(int i = 0; i < nums.length; i++) {
            if(nums[i] != 0) {
                nums[j++] = nums[i];
            }
        }
        //第二次遍历: 非零元素已移动完毕,最后将数组末尾元素赋值为0即可
        for(int i = j; i < nums.length; i++) {
            nums[i] = 0;
        }
    }
}
// @lc code=end
